addons/pos_pinelabs/models/pos_payment_method.py

```diff
@@ -0,0 +1,81 @@
+from odoo import api, fields, models,  _
+from odoo.exceptions import UserError
+
+from .pinelabs_pos_request import PinelabsPosRequest
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    pinelabs_merchant = fields.Char(string='Pine Labs Merchant ID', help='A merchant id issued directly to the merchant by Pine Labs.')
+    pinelabs_store = fields.Char(string='Pine Labs Store ID', help='A store id issued directly to the merchant by Pine Labs.')
+    pinelabs_client = fields.Char(string='Pine Labs Client ID', help='A client id issued directly to the merchant by Pine Labs.')
+    pinelabs_security_token = fields.Char(string='Pine Labs Security Token', help='A security token issued directly to the merchant by Pine Labs.')
+    pinelabs_allowed_payment_modes = fields.Selection(selection=[('all', "All"), ('card', "Card"), ('upi', "Upi")], string='Pine Labs Allowed Payment Modes', help='Accepted payment modes by Pine Labs for transactions.')
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('pinelabs', 'Pine Labs')]
+
+    def pinelabs_make_payment_request(self, data):
+        pinelabs = PinelabsPosRequest(self)
+        body = pinelabs._pinelabs_request_body()
+        body.update({
+            'Amount': data.get('amount'),
+            'TransactionNumber': data.get('transactionNumber'),
+            'SequenceNumber': data.get('sequenceNumber')
+        })
+        response = pinelabs._call_pinelabs(endpoint='UploadBilledTransaction', payload=body)
+        if not response.get('ResponseCode') and response.get('ResponseMessage') == "APPROVED":
+            return {
+                'responseCode': response.get('ResponseCode'),
+                'status': response.get('ResponseMessage'),
+                'plutusTransactionReferenceID': response.get('PlutusTransactionReferenceID'),
```

Comment:
why don't you store this in the db? It looks like you don't actually need it in the client, and only send it there in order to have it for the next call to pinelabs, which is executed on the server anyways

Reply:
We need to store the `plutusTransactionReferenceID` in the database to facilitate refund requests. Currently, no other data related to the payment process needs to be stored. I will store it in the new field `plutus_transaction_reference_id`.

Reply:
maybe just `plutus_transaction_reference`. In the context of odoo, adding `_id` might suggest that it's a relational field